pkg/sdk/sdk.go

```diff
@@ -0,0 +1,152 @@
+package sdk
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	respModel "github.com/PereRohit/util/model"
+	"github.com/gorilla/mux"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"time"
+)
+
+type htmlToPdfSvc struct {
+	svcUrl string
+	client http.Client
+}
+
+func NewHtmlToPdfSvc(url string) HtmlToPdfSvcI {
+	return &htmlToPdfSvc{
+		svcUrl: url,
+		client: http.Client{
+			Timeout: 5 * time.Second,
+		},
+	}
+}
+
+type HtmlToPdfSvcI interface {
+	Register([]byte) (string, error)
+	Replace([]byte, string) error
+	GeneratePdf(map[string]interface{}, string) ([]byte, error)
+}
+
+func (h *htmlToPdfSvc) Register(fileBytes []byte) (string, error) {
+	//take in the slice of bytes so that you can pass it directly to io.copy
+	//take io.reader as argument
+	file := bytes.NewReader(fileBytes)
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+	part, err := writer.CreateFormFile("file", "output")
+	if err != nil {
+		return "", err
+	}
+	_, err = io.Copy(part, file)
+	writer.Close()
+	r, err := h.client.Post(h.svcUrl+"/v1/register", writer.FormDataContentType(), body)
+	if err != nil {
+		return "", err
+	}
+
+	if r.StatusCode < 200 || r.StatusCode > 299 {
+		return "", fmt.Errorf("non success status code received : %v", r.StatusCode)
+	}
+	resp, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return "", err
+	}
+	var response respModel.Response
+	err = json.Unmarshal(resp, &response)
+	if err != nil {
+		return "", err
+	}
+	x := response.Data
+	m, ok := x.(map[string]interface{})
+	if !ok {
+		errNew := errors.New("unable to parse response data")
+		return "", errNew
+	}
+	return fmt.Sprint(m["id"]), err
```

Comment:
Done

Reply:
Not the correct way to check.

What happens if the value for the `id` is a struct? This can happen as the response is asserted to `map[string]interface{}`. This `interface{}` can be anything.

Sprint will try to get the string representation of the struct when the actual expectation is a string.

Can this check be done using type assertion and accordingly check if the `id` field is present?